lib/api/pool.go

```diff
@@ -26,16 +26,36 @@ type PoolMetadata struct {
 }
 
 type PoolSpec struct {
-	IPIP        bool `json:"ipip,omitempty"`
-	NATOutgoing bool `json:"nat-outgoing"`
-	Ipam        bool `json:"ipam"`
-	Disabled    bool `json:"disabled"`
+	// Contains configuration for ipip tunneling
+	// for this pool. If not specified, then ipip
+	// tunneling is disabled for this pool.
+	IPIP *IPIPConfiguration `json:"ipip,omitempty"`
```

Comment:
Making it a pointer is more about making it an optional field in the json.

If this isn't a pointer and it's not specified, your json looks like this:

```
spec:
  ipip: {}
```

If it is a pointer and not specified, you get this:
```
spec:
```

This is also the Kubernetes way of doing "optional" fields.  Any non-mandatory fields are pointers and have `omitempty` on them.